hack/non-olm-install/install-gitops-operator.sh

```diff
@@ -73,72 +81,80 @@ function rollback() {
   if [ ! -z "${PREV_OPERATOR_IMG}" ]; then
     export OPERATOR_IMG=${PREV_OPERATOR_IMG}    
     prepare_kustomize_files
-    ${KUSTOMIZE} build ${TEMP_DIR} | ${KUBECTL} apply -f -
-    echo "[INFO] Upgrade Unsuccessful!!";
+    ${KUSTOMIZE} build ${WORK_DIR} | ${KUBECTL} apply -f -
+    echo "[INFO] Operator update operation was unsuccessful!!";
   else
     echo "[INFO] Installing image for the first time. Nothing to rollback. Quitting..";
   fi
   exit 1;
 }
 
-# deletes the temp directory
+# deletes the work directory
 function cleanup() {
-  rm -rf "${TEMP_DIR}"
-  echo "[INFO] Deleted temp working directory ${TEMP_DIR}"
+  # Check if timeout binary is available in the PATH environment variable
+  timeout=$(which timeout)
+  if [ -z ${timeout} ]; then
+    echo "[INFO] Deleting directory ${WORK_DIR} without timeout"
+    rm -rf "${WORK_DIR}"
+  else
+    # If the command hangs for more than 20 minutes kill it
+    echo "[INFO] Deleting directory ${WORK_DIR} with timeout"
+    timeout 600 rm -rf "${WORK_DIR}"||echo "[ERROR] Directory deletion timed out, please remove it manually"
```

Comment:
nit: comment says 20 mins but timeout appears to be setting the limit as 10 mins

Reply:
Corrected.